File: packages/http-client-csharp/generator/Microsoft.Generator.CSharp.Input/src/InputTypes/InputAuth.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

using System.Text.Json.Serialization;

namespace Microsoft.Generator.CSharp.Input
{
    public class InputAuth
    {
        [JsonConstructor]
        public InputAuth(InputApiKeyAuth? apiKey, InputOAuth2Auth? oAuth2)
        {
            ApiKey = apiKey;
            OAuth2 = oAuth2;
        }

        public InputAuth() : this(null, null) { }

        public InputApiKeyAuth? ApiKey { get; init; }
        public InputOAuth2Auth? OAuth2 { get; init; }
    }
}

File: packages/http-client-csharp/emitter/src/type/input-auth.ts
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License. See License.txt in the project root for license information.

import type { InputApiKeyAuth } from "./input-api-key-auth.js";
import type { InputOAuth2Auth } from "./input-oauth2-auth.js";

export interface InputAuth {
  apiKey?: InputApiKeyAuth;
  oAuth2?: InputOAuth2Auth;
}
